shrine_with_cart/mdc_100_series/lib/short_bottom_sheet.dart

```diff
@@ -0,0 +1,578 @@
+                  : _width - 64,
+              height: _cartHeight,
+              child: Padding(
+                padding: EdgeInsets.only(top: 8.0, bottom: 8.0),
+                child: ProductList(),
+              ),
+            ),
+            ExtraProductsNumber()
+          ]),
+          // Ensures the thumbnails are "pinned" to the top left when opening the
+          // sheet by filling the space beneath them.
+          Expanded(child: Container())
+        ]),
+      ),
+    );
+  }
+
+  Widget _buildShoppingCartPage() {
+    return Opacity(
+        opacity: _cartOpacityAnimation.value, child: ShoppingCartPage());
+  }
+
+  Widget _buildCart(BuildContext context, Widget child) {
+    // numProducts is the number of different products in the cart (does not
+    // include multiple of the same product).
+    AppStateModel model = ScopedModel.of<AppStateModel>(context);
+    int numProducts = model.productsInCart.keys.length;
+    int totalCartQuantity = model.totalCartQuantity;
+
+    _adjustCartPadding(numProducts);
+    _updateWidth(numProducts);
+    _updateAnimations(context);
+
+    return Semantics(
+      button: true,
+      value: "Shopping cart, $totalCartQuantity items",
+      child: Container(
+        width: _widthAnimation.value,
+        height: _heightAnimation.value,
+        child: Material(
+            type: MaterialType.canvas,
+            animationDuration: Duration(milliseconds: 0),
+            shape: BeveledRectangleBorder(
+              borderRadius: BorderRadius.only(
+                  topLeft: Radius.circular(_shapeAnimation.value)),
+            ),
+            elevation: 4.0,
+            color: kShrinePink50,
+            child: _revealCart()
+                ? _buildShoppingCartPage()
+                : _buildThumbnails(numProducts)),
+      ),
+    );
+  }
+
+  // Closes the cart if the cart is open, otherwise exits the app (this should
+  // only be relevant for Android).
+  Future<bool> _onWillPop() {
+    if (_isOpen) {
+      close();
+    } else {
+      SystemNavigator.pop();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    timeDilation = 1.0;
+
+    return AnimatedSize(
+      key: _shortBottomSheetKey,
+      duration: Duration(milliseconds: 225),
+      curve: Curves.easeInOut,
+      vsync: this,
+      alignment: FractionalOffset.topLeft,
+      child: WillPopScope(
+        onWillPop: _onWillPop,
+        child: SlideTransition(
+          position: _slideAnimation,
+          child: GestureDetector(
+            behavior: HitTestBehavior.opaque,
+            onTap: open,
+            child: ScopedModelDescendant<AppStateModel>(
+              builder: (context, child, model) => AnimatedBuilder(
+                    builder: _buildCart,
+                    animation: _controller,
+                  ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class ProductList extends StatefulWidget {
```

Comment:
Since there are other lists of products, come up with another class name. Maybe product thumbnail something.

Reply:
Done! Renamed to `ProductThumbnailRow.`